Update continuous-delivery-faqs.md (#6361)
harness/developer-hub: kb/continuous-delivery/continuous-delivery-faqs.md

- You must enable `Allow users to mark a running Step as failure` in your Harness account's [default settings](/docs/platform/settings/default-settings).
- This feature is behind the feature flag `CDS_MARK_PIPELINE_AS_FAILURE`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

For more refer to the following Harness [documentation](https://developer.harness.io/docs/platform/pipelines/failure-handling/mark-as-failed-pipeline/).

#### How can we fully stop a pipeline if it was aborted but still shows a running status, preventing us from aborting it again?

> Please reach out to [Harness support](support@harness.io) to enable the feature flag `CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE`. Once this flag is enabled, any future pipelines being executed will no longer experience the issue of being stuck in a running status after being aborted

Reviewer: ```suggestion
Please reach out to [Harness Support](mailto:support@harness.io)) to enable the feature flag `CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE`. Once this flag is enabled, any future pipelines being executed will no longer experience the issue of being stuck in a running status after being aborted
```

- You must enable `Allow users to mark a running Step as failure` in your Harness account's [default settings](/docs/platform/settings/default-settings).
- This feature is behind the feature flag `CDS_MARK_PIPELINE_AS_FAILURE`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

For more refer to the following Harness [documentation](https://developer.harness.io/docs/platform/pipelines/failure-handling/mark-as-failed-pipeline/).

#### The pipeline was aborted, however, the status still shows running.  It is not allowing us to abort the pipeline.  What can be done to completely stop the pipeline?

> Contact Harness support to enable the feature flag CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE.  Once this is enabled, any future pipelines being executed will no longer experience this issue.

Reviewer: ```suggestion
Please reach out to [Harness support](support@harness.io) to enable the feature flag `CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE`. Once this flag is enabled, any future pipelines being executed will no longer experience the issue of being stuck in a running status after being aborted
```

> #### The feature flag CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE was enabled to prevent executed pipelines from staying in a running state even though it was aborted.  However, retrying the failed execution of the pipeline is still showing running after being aborted.  Why does the aborted retried execution still show running?

Reviewer: ```suggestion
#### Why, despite the activation of the feature flag `CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE`, does the retried execution of an aborted pipeline persistently display a running status?
```

> The CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE feature flag, once enabled, will only work for any pipelines being executed.  However, it will not work on executions that are being retried as the feature flag was not enabled for that retried execution.

Reviewer: ```suggestion
Enabling the `CDS_FIX_RETRY_FAILED_PIPELINE_WITH_USE_SERVICES_FROM_STAGE` feature flag only affects the execution of new pipelines. It doesn't apply to retried executions since the flag isn't active for those retries.
```

> #### When running a pipeline, should steps be skipped when the user selects 'skip pre-flight check' under their service and infrastructure?

Reviewer: ```suggestion
#### Should steps be skipped when a user selects `skip pre-flight check` under their service and infrastructure during pipeline execution?
```

> Preflight checks run before the pipeline execution starts. If those fail, the pipeline execution won’t even start. Any error seen is part of the service step which is part of pipeline execution and not part of preflight checks.  Preflight checks only validate the Pipeline inputs and connectors referenced in the pipeline, but not the artifacts and manifests in the service.

Reviewer: ```suggestion
Preflight checks occur before the pipeline execution begins, ensuring the validity of pipeline inputs and connectors. If these checks fail, the pipeline won't initiate. It's important to note that any errors encountered during the pipeline execution are associated with the service step within the pipeline, not the preflight checks. Additionally, preflight checks do not validate artifacts and manifests within the service.
```

#### Is there a way for Harness to deploy the latest tag version published to our artifact server?

> Harness can utilize the expression <+lastPublished.tag> in order to obtain the latest tag version available on the artifact server.  For more information on how to utilize this expression, please visit https://developer.harness.io/kb/continuous-delivery/articles/last-published-tag/.

Reviewer: ```suggestion
Yes, in Harness, you can deploy the latest tag version to your artifact server. By leveraging the `<+lastPublished.tag>` expression, Harness can fetch the most recent tag version available on the artifact server. For comprehensive instructions on implementing this approach, please consult the following Harness [documentation](https://developer.harness.io/kb/continuous-delivery/articles/last-published-tag/)
```

> #### How can we rerun a given pipeline execution similar to using the "retry from failed stage" option from the UI?

Reviewer: ```suggestion
#### What is the equivalent method to rerun a specific pipeline execution, resembling the functionality of selecting "retry from failed stage" from the UI?
```

> Users can utilize the following API to retry a failed stage: https://apidocs.harness.io/tag/Pipeline-Execute#operation/retryPipeline
> 
> Users can also provide the input data set required to rerun the failed pipeline.

Reviewer: ```suggestion
Users can utilize the following API to retry a failed stage: [retryPipeline](https://apidocs.harness.io/tag/Pipeline-Execute#operation/retryPipeline). Additionally, users can supply the necessary input data set required to rerun the failed pipeline.
```

> #### When trying to import a .yaml file from GitHub to create a new pipeline, I get the message "This file is already in use by another pipeline". Since we don't have any other pipelines in this project, is there is a duplicate entry somewhere that I may not be aware about?

Reviewer: ```suggestion
#### When attempting to import a **.yaml file** from GitHub to create a new pipeline, the message `This file is already in use by another pipeline` is displayed. Given that there are no other pipelines in this project, is there a possibility of a duplicate entry that I may not be aware of?
```

> There may be two pipeline entities in the database with the file path in question associated with the Harness account and with the GitHub url. Attempting to import the file again triggers the 'File Already Imported' pop-up on the screen.
> Additionally, users have the option to bypass this check by clicking the 'Import' button again.

Reviewer: ```suggestion
It's possible that there are two pipeline entities in the database, each linked to the same file path from the Harness account and the GitHub URL. Trying to import the file again may trigger the `File Already Imported` pop-up on the screen. However, users can choose to bypass this check by clicking the `Import` button again.
```
